tests/unit/acquisition/test_optimizer.py

```diff
@@ -59,6 +67,22 @@ def test_generate_random_search_optimizer_raises_with_invalid_sample_size() -> N
         generate_random_search_optimizer(num_samples=-5)
 
 
+@pytest.mark.parametrize("batch_size", [0, -2])
+def test_optimize_discrete_raises_with_invalid_vectorized_batch_size(batch_size: int) -> None:
+    search_space = DiscreteSearchSpace(tf.constant([[-0.5], [0.2], [1.2], [1.7]]))
+    acq_fn = _quadratic_sum([1.0])
+    with pytest.raises(ValueError):
+        optimize_discrete(search_space, (acq_fn, batch_size))
+
+
+@pytest.mark.parametrize("batch_size", [0, -2])
+def test_random_optimizer_raises_with_invalid_vectorized_batch_size(batch_size: int) -> None:
+    search_space = Box([-1], [2])
+    acq_fn = _quadratic_sum([1.0])
+    with pytest.raises(ValueError):
+        generate_random_search_optimizer()(search_space, (acq_fn, batch_size))
```

Comment:
The mypy error here is actually a preexisting issue: generate_random_search_optimizer (added in #242) has a return type of `AcquisitionOptimizer[SP]` (which doesn't make sense). It should instead be `AcquisitionOptimizer[SearchSpace]`.

Reply:
(you'll need to change the type of space on line 605 as well)